app/controllers/concerns/valid_request.rb

```diff
@@ -34,6 +34,8 @@ def _compute_redirect_to_location(request, options) # :nodoc:
       "#{URL.protocol || request.protocol}#{request.host_with_port}#{options}"
     when Proc
       _compute_redirect_to_location request, instance_eval(&options)
+    when Array
+      url_for
```

Comment:
I'm not sure, but I wonder if this should match the line below?

```suggestion
      url_for(options)
```

Reply:
It doesn't work because `url_for` is [not designed to work with Arrays]( https://api.rubyonrails.org/v7.0.4/classes/ActionDispatch/Routing/UrlFor.html).

I was running into problems, which is why I think it needs to be explicitly added here.

Reply:
Oh, weird. That makes me suspicious of `Rack::Attack` -- or maybe we've configured it in an unusual way? Either way, if this gets this unstuck it's good :+1:

Reply:
I think it has to be one of our controllers or omniauth. Either way, the blank `url_for` fallbacks takes the user to the homepage. I will draft up a issue for follow up.